csrc/ir/nodes.cpp

```diff
@@ -2520,21 +2535,6 @@ std::pair<IterDomain*, IterDomain*> IterDomain::split(
   TORCH_CHECK(
       factor->isIntegralScalar(), "Cannot split by non-integer value ", factor);
 
-  if (factor->getValType() == ValType::Others) {
```

Comment:
Why this part is removed?

Reply:
I don't think this restriction makes sense. For example, the split size can be `castOp(DataType::Index, 5)` and it works totally fine. I don't see why fusion inputs are special than a general `Val*`.

Reply:
I thought there was some constraint about split factors. It may be that having an arbitrary val as a factor would mean that that split tensor would have a dependency with the val and its defining expression, but I don't think we consider it as a dependency, for example, when sorting expressions. We may not see any problem as I think we tend to put scalar expressions before tensor expressions, but I wonder if it's really guaranteed to be safe.

Reply:
My understanding for this is, the split factor will only be used through the extent of the inner ID. For this use, I don't think this will be an issue, because when we codegen a `Val*` that is not fusion input, we automatically codegen its entire definition, instead of just its name. What makes me feel worried is segmentation, because of the lack of dependency tracking, the segmenter might not correctly assign the needed inputs to the fusion. But I think this problem exist even if the split factor is a fusion input of the unsegmented fusion. I can not foresee any other issue.

Reply:
IIRC, we don't segment scalar ops but just include all scalar ops in every segment. Let's remove the check and revisit if necessary.

Reply:
No need this time, but would appreciate if this change were done as a separate PR.